### PR TITLE
[LW] Add testing around edge cases when committing transactions

### PR DESCRIPTION
**Pull request by Jolyon-S**
**Goals (and why)**:
https://github.com/palantir/atlasdb/pull/5637 was causing validation failures - not the ABA case, but rather AB where the B write occurred at commit timestamp plus one. Aim here is to add testing around this case to confirm that this is also no longer an issue with validation here.

**Implementation Description (bullets)**:
Add a test that fakes a transaction committing at `commitTimestamp + 1`.

**Concerns (what feedback would you like?)**:
Is this test reasonable to understand?

**Where should we start reviewing?**:
Diff

**Priority (whenever / two weeks / yesterday)**:
This week

### CODE REVIEW

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java [hunk 1]
```diff
@@ -374,12 +386,85 @@ public void getRowsUsesCacheAsExpected() {
                     txn,
                     ImmutableSet.of(
                             CellReference.of(TABLE_REF, CELL_2), CellReference.of(TABLE_REF, CELL_3), TABLE_CELL_4));
-            // we weren't able to cache our own write so we look this up
+            // we weren't able to cache our own write, so we look this up
```
Comment (Jolyon-S) [inline timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java:389]: grammar

**Review by jeremyk-91 — COMMENTED** [review]
Thanks for opening this PR! While this test does cover what we want to do (it builds on the previous tests by integrating the LW layer), there is an issue. Specifically, pre-commit conditions are run _after_ the serializable conflict check, and so this test still passes even if I undo the commit + 1 fix. I think we need to do the writes slightly earlier.

**Review by jeremyk-91 — APPROVED** [review]
